sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -126,19 +149,47 @@ class SQLQueryTestSuite extends QueryTest with SharedSQLContext {
         sql + "\n" +
         s"-- !query $queryIndex schema\n" +
         schema + "\n" +
-         s"-- !query $queryIndex output\n" +
+        s"-- !query $queryIndex output\n" +
         output
     }
   }
 
+  /** A test case. */
+  sealed trait TestCase {
+    val name: String
+    val inputFile: String
+    val resultFile: String
+  }
+
+  /** A regular test case. */
+  private case class RegularTestCase(
+      name: String, inputFile: String, resultFile: String) extends TestCase
+
+  /** A UDF test case. */
+  private case class UDFTestCase(
+      name: String, inputFile: String, resultFile: String, udf: TestUDF) extends TestCase
+
   private def createScalaTestCase(testCase: TestCase): Unit = {
     if (blackList.exists(t =>
         testCase.name.toLowerCase(Locale.ROOT).contains(t.toLowerCase(Locale.ROOT)))) {
       // Create a test case to ignore this case.
       ignore(testCase.name) { /* Do nothing */ }
-    } else {
-      // Create a test case to run this case.
-      test(testCase.name) { runTest(testCase) }
+    } else testCase match {
+      case UDFTestCase(_, _, _, udf: TestPythonUDF) if !shouldTestPythonUDFs =>
+        ignore(s"${testCase.name} is skipped because " +
```

Comment:
I am afraid these tests might not be triggered in our test environments. To ensure it works as expected, we should throw an error instead of logging a message.

I think we can provide a new environment variable, IGNORE_SKIPPED_TESTS. If the env is not set, we should throw an exception. In the exception message, we can clearly explain the workaround by setting an environment variable IGNORE_SKIPPED_TESTS.

Reply:
The problem is that it might break other CI running, for instance in other venders and it will force them to set  `IGNORE_SKIPPED_TESTS`. Currently, PySpark tests are being skipped in the cases (which is common in Python projects as far as I know)

If we should add a configuration like `IGNORE_SKIPPED_TESTS`, I would like to do that separately for PySpark side as well. But I and some committers and people as far as I know tend to ignore them for now.